### PR TITLE
refactor: postUpgradeCommandsExecutor

### PR DESCRIPTION
**Pull request by damien-danglard**
## Changes

Extract `postUpgradeCommandsExecutor` content into `upgradeTaskExecutor`.

## Context

This refactoring is about prepare PR of create `preUpgradeCommandsExecutor`.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/update/branch/execute-upgrade-commands.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  const artifactErrors = [];
+  if (allowedUpgradeCommands.some((pattern) => regEx(pattern).test(cmd))) {
+    try {
+      const compiledCmd = allowUpgradeCommandTemplating
+        ? compile(cmd, mergeChildConfig(config, upgrade))
+        : cmd;
+
+      logger.trace(
+        { cmd: compiledCmd },
+        'Executing ' + taskType.toLowerCase() + ' task'
```
Comment (viceice) [inline lib/workers/repository/update/branch/execute-upgrade-commands.ts:112]:
```suggestion
        `Executing ${taskType.toLowerCase()} task'
```
use string template

File: lib/workers/repository/update/branch/execute-upgrade-commands.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      );
+      const execResult = await exec(compiledCmd, {
+        cwd: GlobalConfig.get('localDir'),
+      });
+
+      logger.debug(
+        { cmd: compiledCmd, ...execResult },
+        'Executed ' + taskType.toLowerCase() + ' task'
```
Comment (viceice) [inline lib/workers/repository/update/branch/execute-upgrade-commands.ts:120]: same

File: lib/workers/repository/update/branch/execute-upgrade-commands.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        stderr: sanitize(error.message),
+      });
+    }
+  } else {
+    logger.warn(
+      {
+        cmd,
+        allowedUpgradeCommands,
+      },
+      taskType + ' task did not match any on allowedPostUpgradeCommands list'
```
Comment (viceice) [inline lib/workers/repository/update/branch/execute-upgrade-commands.ts:134]: same

File: lib/workers/repository/update/branch/execute-upgrade-commands.ts [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+  updatedArtifacts: FileChange[],
+  taskType: string
+): Promise<FileChange[]> {
+  const status = await getRepoStatus();
+  let updatedUpdatedArtifacts = updatedArtifacts;
+
+  for (const relativePath of status.modified.concat(status.not_added)) {
+    for (const pattern of fileFilters) {
+      if (minimatch(relativePath, pattern)) {
+        logger.debug({ file: relativePath, pattern }, taskType + ' file saved');
```
Comment (viceice) [inline lib/workers/repository/update/branch/execute-upgrade-commands.ts:157]: same

File: lib/workers/repository/update/branch/execute-upgrade-commands.ts [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+      }
+    }
+  }
+
+  for (const relativePath of status.deleted || []) {
+    for (const pattern of fileFilters) {
+      if (minimatch(relativePath, pattern)) {
+        logger.debug(
+          { file: relativePath, pattern },
+          taskType + ' file removed'
```
Comment (viceice) [inline lib/workers/repository/update/branch/execute-upgrade-commands.ts:184]: same

File: lib/workers/repository/update/branch/execute-upgrade-commands.ts [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+  addMeta({ dep: upgrade.depName });
+  logger.trace(
+    {
+      tasks: upgradeTask,
+      allowedCommands: allowedUpgradeCommands,
+    },
+    `Checking for ${taskType.toLowerCase()} tasks`
+  );
+
+  const commands = upgradeTask?.commands ?? [];
```
Comment (viceice) [inline lib/workers/repository/update/branch/execute-upgrade-commands.ts:47]: we should validate allowed commands here, so we can skip file persisting if no allowed commands are available.

File: lib/workers/repository/update/branch/execute-post-upgrade-commands.ts [hunk 7]
```diff
@@ -1,163 +1,41 @@
...
-            // If the file is created or modified by a previous post-update command, remove the modification from updatedArtifacts
-            updatedArtifacts = updatedArtifacts.filter(
-              (ua) => !(ua.type === 'addition' && ua.path === relativePath)
-            );
-          }
-        }
-      }
-    }
+    updatedArtifacts = result.updatedArtifacts;
+    artifactErrors = [...artifactErrors, ...result.artifactErrors];
```
Comment (viceice) [inline lib/workers/repository/update/branch/execute-post-upgrade-commands.ts:31]: we should pass the array to the called function and add results directly, so we don't need to clone it in each iteration.

File: lib/workers/repository/update/branch/execute-upgrade-commands.ts [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+    for (const cmd of commands) {
+      const commandError = await upgradeCommandExecutor(
+        allowedUpgradeCommands ?? [],
+        cmd,
+        allowUpgradeCommandTemplating,
+        config,
+        upgrade,
+        taskType
+      );
+      artifactErrors = [...artifactErrors, ...commandError];
```
Comment (viceice) [inline lib/workers/repository/update/branch/execute-upgrade-commands.ts:86]: use push instead, so we don't clone the array on each iteration .

i think we should better pass the existing array and push in called function to have less clones

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by rarkins** [comment]
Closing due to inactivity. Reopening or re-raising are welcome!

**Pull request closed without merging**